engine/src/witness/sol.rs

```diff
@@ -104,6 +106,36 @@ impl VoterApi<SolanaIngressTracking> for SolanaIngressTrackingVoter {
 	}
 }
 
+#[derive(Clone)]
+struct SolanaNonceTrackingVoter {
+	client: SolRetryRpcClient,
+}
+
+#[async_trait::async_trait]
+impl VoterApi<SolanaNonceTracking> for SolanaNonceTrackingVoter {
+	async fn vote(
+		&self,
+		_settings: <SolanaNonceTracking as ElectoralSystem>::ElectoralSettings,
+		properties: <SolanaNonceTracking as ElectoralSystem>::ElectionProperties,
+	) -> Result<<<SolanaNonceTracking as ElectoralSystem>::Vote as VoteStorage>::Vote, anyhow::Error>
+	{
+		let (nonce_account, previous_nonce) = properties;
+		let (response_account, response_nonce) =
+			nonce_witnessing::get_durable_nonces(&self.client, vec![nonce_account])
```

Comment:
I'd really prefer to pass all nonce accounts instead of one by one. This way we can take advantage of the `getMultipleAccounts` and use 1 RPC call to query the state of all 10 nonces instead of 10 separate RPC calls.

Reply:
https://linear.app/chainflip/issue/PRO-1588/witness-all-nonces-at-once